homeassistant/components/zwave_js/light.py

```diff
@@ -499,61 +542,137 @@ def __init__(
         """Initialize the light."""
         super().__init__(config_entry, driver, info)
 
-        self._last_color: dict[str, int] | None = None
-        self._supported_color_modes.discard(ColorMode.BRIGHTNESS)
+        self._last_on_color: dict[ColorComponent, int] | None = None
+        self._last_brightness: int | None = None
 
     @property
-    def brightness(self) -> int:
-        """Return the brightness of this light between 0..255."""
-        return 255
+    def brightness(self) -> int | None:
+        """Return the brightness of this light between 0..255.
 
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if device is on (brightness above 0)."""
+        Z-Wave multilevel switches use a range of [0, 99] to control brightness.
+        """
         if self.info.primary_value.value is None:
             return None
-        return any(value != 0 for value in self.info.primary_value.value.values())
+        if self._target_brightness and self.info.primary_value.value is False:
+            # Binary switch exists and is turned off
+            return 0
+
+        # Brightness is encoded in the color channels by scaling them lower than 255
+        color_values = [v.value for v in self._get_color_values() if v is not None]
+        max_value = max([v for v in color_values if v is not None])
```

Comment:
This will error if the list is empty.

```py
Python 3.12.4 (main, Jun  8 2024, 18:29:57) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.22.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: max([])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[1], line 1
----> 1 max([])

ValueError: max() iterable argument is empty
```

If we know that this light always supports red, green and blue colors, ie we don't have to worry about an empty list here, it's weird that we check `self._supports_color` in this class.

Reply:
The modified discovery scheme creates an instance of this class for
a) devices that only have a current color, but no switching capabilities
b) devices that have a binary switch and a current color

So there should always be at least one color component, otherwise the scheme wouldn't match.

> it's weird that we check self._supports_color in this class.

Due to the discovery scheme, it might be that we end up with a white-spectrum light that cannot be dimmed, but turned on and off. I'm not sure how to match lights that have exactly RGB support, but nothing else, since the discovery scheme seems to match when ANY of the specified values are there, not ALL.

Should the initial `if` in `async_turn_on` and `brightness` should defer to the parent class if there is no RGB support? That way we should avoid this problem altogether.